arangod/Transaction/Methods.cpp

```diff
@@ -1825,19 +1862,19 @@ Future<OperationResult> transaction::Methods::truncateLocal(std::string const& c
             OperationResult(TRI_ERROR_CLUSTER_SHARD_LEADER_REFUSES_REPLICATION, options));
       }
       switch (followerInfo->allowedToWrite()) {
-      case FollowerInfo::WriteState::FORBIDDEN:
-        // We cannot fulfill minimum replication Factor. Reject write.
-        return OperationResult(TRI_ERROR_ARANGO_READ_ONLY, options);
-      case FollowerInfo::WriteState::STARTUP:
-        return OperationResult(TRI_ERROR_CLUSTER_BACKEND_UNAVAILABLE, options);
-      default:
-        break;
+        case FollowerInfo::WriteState::FORBIDDEN:
+          // We cannot fulfill minimum replication Factor. Reject write.
+          return OperationResult(TRI_ERROR_ARANGO_READ_ONLY, options);
+        case FollowerInfo::WriteState::STARTUP:
+          return OperationResult(TRI_ERROR_CLUSTER_BACKEND_UNAVAILABLE, options);
+        default:
+          break;
       }
 
       // fetch followers
+      replicationType = ReplicationType::LEADER;
       followers = followerInfo->get();
```

Comment:
What bug was fixed here by setting `replicationType` to `Leader` even when there are no followers?

Reply:
No bug is fixed by this. However, truncate did this differently than insert/update/replace/remove:
In truncate, `replicationType` was set to `LEADER` only if there is at least one follower.
In the other write operations, `replicationType` was set to `LEADER` even if there are 0 followers. Now it will be at least consistent.